app/src/main/java/com/github/se/orator/utils/FirestoreHelperMethods.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.se.orator.utils
+
+import android.util.Log
+import com.github.se.orator.model.speaking.InterviewContext
+import com.github.se.orator.model.speechBattle.BattleStatus
+import com.github.se.orator.model.speechBattle.EvaluationResult
+import com.github.se.orator.model.speechBattle.SpeechBattle
+import com.github.se.orator.ui.network.Message
+
+/**
+ * Converts a Firestore map to a SpeechBattle object.
+ *
+ * @param data The map containing SpeechBattle data.
+ * @return A SpeechBattle object, or null if conversion fails.
+ */
+fun mapToSpeechBattle(data: Map<String, Any>): SpeechBattle? {
+  return try {
+    val battleId = data["battleId"] as? String ?: return null
+    val challenger = data["challenger"] as? String ?: return null
+    val opponent = data["opponent"] as? String ?: return null
```

Comment:
I see your point, but I believe using null values is better in this case because it allows us to filter out invalid entries using Kotlin's built-in functions. This helps avoid inconsistencies and reduces the need for extra checks beyond the null checks already available in Kotlin.